replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/HeartbeaterAndVerifierImpl.java

```diff
@@ -73,23 +135,307 @@ public void cleanUp() {
 
   @Override
   public void verify(ReplicationSite site) {
-    final String url = site.getUrl();
+    if (site.getVerifiedUrl() == null) {
+      final String url = site.getUrl();
+      HeartbeatResult heartbeatResult = doDiscovery(url);
 
-    // For now, fake the verification until implemented
-    site.setVerifiedUrl(url);
-    siteManager.save(site);
+      if (heartbeatResult.isHeartbeatSupported()) {
+        site.setRemoteManaged(true);
+        site.setVerifiedUrl(url);
+        siteManager.save(site);
+      } else if (heartbeatResult.isTmpRedirect()) {
+        HeartbeatResult redirectResult = doDiscovery(heartbeatResult.getRedirectUrl());
```

Comment:
if the initial heartbeat call is successful and give us a new url do we need to re-run discovery at that point? Seems like we have verified it is an ION system at that point so we should just set the site properties and save it regardless of whether or not we can verify the redirect.